sql/parse/parse.go

```diff
@@ -426,9 +413,63 @@ func exprToExpression(e sqlparser.Expr) (sql.Expression, error) {
 		}
 
 		return expression.NewOr(lhs, rhs), nil
+	case sqlparser.ValTuple:
+		var exprs = make([]sql.Expression, len(v))
+		for i, e := range v {
+			expr, err := exprToExpression(e)
+			if err != nil {
+				return nil, err
+			}
+			exprs[i] = expr
+		}
+		return expression.NewTuple(exprs...), nil
 	}
 }
 
+func convertVal(v *sqlparser.SQLVal) (sql.Expression, error) {
+	switch v.Type {
+	case sqlparser.StrVal:
+		return expression.NewLiteral(string(v.Val), sql.Text), nil
+	case sqlparser.IntVal:
+		val, err := strconv.ParseInt(string(v.Val), 10, 64)
+		if err != nil {
+			return nil, err
+		}
+		return expression.NewLiteral(val, sql.Int64), nil
+	case sqlparser.FloatVal:
+		val, err := strconv.ParseFloat(string(v.Val), 64)
+		if err != nil {
+			return nil, err
+		}
+		return expression.NewLiteral(val, sql.Float64), nil
+	case sqlparser.HexNum:
+		v := strings.ToLower(string(v.Val))
+		if strings.HasPrefix(v, "0x") {
+			v = v[2:]
+		} else if strings.HasPrefix(v, "x") {
+			v = strings.Trim(v[1:], "'")
+		}
+
+		val, err := strconv.ParseInt(v, 16, 64)
+		if err != nil {
+			return nil, err
+		}
+		return expression.NewLiteral(val, sql.Int64), nil
+	case sqlparser.HexVal:
+		val, err := v.HexDecode()
+		if err != nil {
+			return nil, err
+		}
+		return expression.NewLiteral(val, sql.Blob), nil
+	case sqlparser.ValArg:
+		return expression.NewLiteral(string(v.Val), sql.Text), nil
+	case sqlparser.BitVal:
+		return expression.NewLiteral(v.Val[0] == '1', sql.Boolean), nil
+	}
+
+	panic(fmt.Errorf("unreachable: invalid SQLVal of type: %d", v.Type))
```

Comment:
we should return the error instead of panic here.

Reply:
Can change it to the error return, but the panic was intentional because that part should not be reachable

Reply:
should not, but can be reached, and an error is better than a panic if possible, in mi opinion.

Reply:
Can't because there are no other types implemented for SQLVal unless you set a value that's not correct on a node (which won't happen because they're returned by vitess and we control vitess versioning). But I will change it to an error, anyway.